lib/analyse.ml

```diff
@@ -163,44 +161,51 @@ module Analysis = struct
     | Error (`Msg msg) -> Fmt.error_msg "Error from solver: %s" msg
 
   let of_dir ~solver ~job ~platforms ~opam_repository_commit dir =
-    let solve = solve ~opam_repository_commit ~job ~solver in
-    let ty = type_of_dir dir in
-    let cmd = "", [| "find"; "."; "-maxdepth"; "3"; "-name"; "*.opam" |] in
-    Current.Process.check_output ~cwd:dir ~cancellable:true ~job cmd >>!= fun output ->
     let opam_files =
-      String.split_on_char '\n' output
-      |> List.sort String.compare
-      |> List.filter_map (function
-          | "" -> None
-          | path ->
-            let path =
-              if Astring.String.is_prefix ~affix:"./" path then
-                Astring.String.with_range ~first:2 path
-              else path
-            in
-            let consider_opam_file path =
-              match Fpath.v path |> Fpath.segs with
-              | [_file] -> true
-              | segs when List.exists is_test_dir segs ->
-                Current.Job.log job "Ignoring test directory %S" path;
-                false
-              | _ -> true
-            in
-            let full_path = Filename.concat (Fpath.to_string dir) path in
-            if is_empty_file full_path then (
-              Current.Job.log job "WARNING: ignoring empty opam file %S" path;
-              None
-            ) else if consider_opam_file path then
-              Some path
-            else None
-        )
+      let opam_files full_path =
+        let segs = Fpath.segs full_path |> List.tl in
```

Comment:
Are you sure this will never be an empty list?

Reply:
I think so, the full path returned by `Bos.OS.Path.fold` seems to always start with `./` which is the first segment that I'm dropping. I'll replace it with a check, I can never be so sure.